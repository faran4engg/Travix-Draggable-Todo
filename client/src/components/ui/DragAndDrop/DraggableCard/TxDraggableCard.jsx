import React, { useState } from "react";
import { Draggable } from "react-beautiful-dnd";
import { connect } from "react-redux";
import styled from "styled-components";
import PropTypes from "prop-types";

import TxForm from "../../Form/TxForm";
import TxCard from "../../Card/TxCard";

import { deleteCard } from "../../../../redux/actions";

const Div = styled.div`
  margin: 1em 0 !important;
`;

const TxDraggableCard = props => {
  const { cardData, index, listName } = props;
  const [isEditMode, setIsEditMode] = useState(false);

  const handleEdit = () => {
    setIsEditMode(true);
  };

  const handleDelete = (listName, cardId) => {
    props.dispatch(deleteCard(listName, cardId));
  };

  return isEditMode ? (
    <TxForm
      editData={cardData}
      onCancel={() => setIsEditMode(false)}
      listName={listName}
    />
  ) : (
    <Draggable draggableId={cardData.id} index={index}>
      {provided => (
        <Div
          {...provided.draggableProps}
          {...provided.dragHandleProps}
          ref={provided.innerRef}
        >
          <TxCard
            cardData={cardData}
            onEdit={handleEdit}
            canModify={listName !== "done"}
            onDelete={() => handleDelete(listName, cardData.id)}
          />
        </Div>
      )}
    </Draggable>
  );
};

TxDraggableCard.propTypes = {
  cardData: PropTypes.object,
  index: PropTypes.number,
  listName: PropTypes.string,
  dispatch: PropTypes.func
};

export default connect()(TxDraggableCard);
