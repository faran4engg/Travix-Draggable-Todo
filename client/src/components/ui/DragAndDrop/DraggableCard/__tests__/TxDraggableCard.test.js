import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

import TxDraggableCard from "../TxDraggableCard";

describe("TxDraggableCard", () => {
  it("should render correctly ", () => {
    const store = configureStore()();
    const component = shallow(
      <Provider store={store}>
        <TxDraggableCard
          key="123"
          cardData={{
            title: "Sample Card",
            description: "sample description",
            id: "a1b2c3"
          }}
          listName="Todo"
          index={0}
        />
      </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
