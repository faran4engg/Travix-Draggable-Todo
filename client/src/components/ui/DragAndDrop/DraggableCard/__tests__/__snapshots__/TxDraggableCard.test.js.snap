// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxDraggableCard should render correctly  1`] = `ShallowWrapper {}`;
